packages/trip-form/src/MetroModeSelector/index.tsx

```diff
@@ -0,0 +1,407 @@
+  ${invisibleCss}
+`;
+
+const ModeBar = styled.fieldset`
+  border: none;
+  display: inline-flex;
+  gap: 0 3px;
+  margin: 0 4px 0 0;
+  padding: 0;
+
+  /* <legend> is not shown visually. */
+  & > legend {
+    ${invisibleCss}
+  }
+`;
+
+const accentColor = "#084c8d";
+const activeHoverColor = "#0e5faa";
+
+const ModeButtonWrapper = styled.span<{ fillModeIcons?: boolean }>`
+  position: relative;
+
+  & > label {
+    background: #fff;
+    border-radius: 5px;
+    border: 2px solid ${accentColor};
+    cursor: pointer;
+    display: inline-flex;
+    padding: 0.75rem 0.75rem;
+    transition: all 250ms cubic-bezier(0.27, 0.01, 0.38, 1.06);
+    user-select: none;
+    justify-content: center;
+    aspect-ratio: 1/1; /* stylelint-disable-line property-no-unknown */
+  }
+
+  &:not(:last-of-type) > label {
+    border-bottom-right-radius: 0;
+    border-top-right-radius: 0;
+  }
+  &:not(:first-of-type) > label {
+    border-bottom-left-radius: 0;
+    border-top-left-radius: 0;
+  }
+  & > label:hover {
+    background: #eee;
+    border-color: ${activeHoverColor};
+    box-shadow: 0 5px 15px rgba(0, 0, 0, 0.05),
+      0 4px 10px rgba(0, 123, 255, 0.25);
+  }
+
+  & > input {
+    ${invisibleCss}
+    /* Firefox will still render (tiny) controls, even if their bounds are empty,
+       so move them out of sight. */
+    left: -20px;
+    position: absolute;
+  }
+
+  & > button {
+    ${invisibleCss}
+    background: none;
+    border: none;
+    /* Lateral position is offset so that the button outline is visible when focused. */
+    bottom: 0;
+    left: 4px;
+    position: absolute;
+    right: 4px;
+  }
+
+  & > button:focus {
+    clip: initial;
+    height: initial;
+    width: initial;
+  }
+
+  & > input:checked + label {
+    background: ${accentColor};
+  }
+
+  & > input:checked + label,
+  & > input:checked ~ button {
+    color: white;
+    fill: ${props =>
+      props.fillModeIcons === false ? "inherit" : "currentcolor"};
+  }
+
+  & > input:focus + label {
+    outline: 5px auto blue;
+    /* This next line enhances the visuals in Chromium (webkit) browsers */
+    outline: 5px auto -webkit-focus-ring-color;
+    /* Render the focus outline inside and distinct from the border for both Chrome and Firefox. */
+    outline-offset: -4px;
+  }
+
+  & > input:checked + label:hover {
+    background: ${activeHoverColor};
+  }
+
+  & > label > svg {
+    color: ${accentColor};
+    display: inline-block;
+    height: 32px;
+    margin: auto;
+    vertical-align: middle;
+    width: 32px;
+    fill: ${props =>
+      props.fillModeIcons === false ? "inherit" : "currentcolor"};
+  }
+
+  & > input:checked + label > svg {
+    color: #eee;
+  }
+`;
+
+const HoverPanel = styled.div`
+  min-width: 300px;
+  padding: 0 10px;
+  width: 75%;
+  z-index: 10;
+`;
+
+const HoverInnerContainer = styled.div`
+  background: #fff;
+  border-radius: 4px;
+  box-shadow: 0 5px 15px rgba(0, 0, 0, 0.05), 0 4px 10px rgba(0, 123, 255, 0.25);
```

Comment:
This is a bit intense we should probably increase the spread and lower the opacity -- also can we have the color match the button color?

Reply:
Tuned it down and made it match!